src/wasm/wasm-binary.cpp

```diff
@@ -2243,7 +2243,14 @@ void WasmBinaryReader::verifyInt64(int64_t x) {
 void WasmBinaryReader::readHeader() {
   BYN_TRACE("== readHeader\n");
   verifyInt32(BinaryConsts::Magic);
-  verifyInt32(BinaryConsts::Version);
+  auto version = getInt32();
+  if (version != BinaryConsts::Version) {
+    if (version == 0x1000d) {
+      throwError("this looks like a wasm component, which Binaryen does not "
```

Comment:
We can change `wasm` to `Wasm` in the error string, and it could be nice to give a link to #6728. It allows people to follow the progress and maybe help if they're interested.

Reply:
Good idea, I added a link now.

We use lowercase `wasm` which is more informal in our other error strings, so for consistency I didn't change that. Though I agree in proper English it should be capitalized.